File: 0077-combinations/0077-combinations.java
class Solution {
    
    List<List<Integer>> res = new ArrayList<>();
    
    public void help(int index , int n , int k , ArrayList<Integer> path){
        
        if(k == 0){
            res.add(new ArrayList<Integer>(path));
            return;
        }
        
        
        for(int i = index;i<=n;i++){
            path.add(i);
            help(i+1,n,k-1,path);
            path.remove(path.size()-1);
        }
        
    }
    
    public List<List<Integer>> combine(int n, int k) {
        int index = 1;
        ArrayList<Integer>path = new ArrayList<>();
        help(index,n,k,path);
        
        return res;
    }
}

File: 0077-combinations/0077-combinations.cpp
class Solution {
public:
    // _ _
    vector<vector<int>>res;
    void help(int n , int index , int k , vector<int>&path){
        if(k==0){
            res.push_back(path);
            return;
        }
        // if(index > n){
        //     return;
        // }
        
        for(int i=index;i<=n;i++){
            path.push_back(i);
            //cout<<i<<" ";
            help(n,i+1,k-1,path);
            path.pop_back();
        }
        
        
    }
    vector<vector<int>> combine(int n, int k) {
        vector<int>path;
        help(n,1,k,path);
        return res;
    }
};